src/native/corehost/hostmisc/pal.windows.cpp

```diff
@@ -423,7 +423,7 @@ bool pal::get_global_dotnet_dirs(std::vector<pal::string_t>* dirs)
 
 // To determine the OS version, we are going to use RtlGetVersion API
 // since GetVersion call can be shimmed on Win8.1+.
-typedef NTSTATUS (WINAPI *pFuncRtlGetVersion)(RTL_OSVERSIONINFOW *);
+typedef LONG (WINAPI *pFuncRtlGetVersion)(RTL_OSVERSIONINFOW *);
```

Comment:
I did not want to include `windows.h` for this. Host include that.

Reply:
I am not sure I understand this - the pal.windows.cpp set of includes didn't seem to have changed.

Reply:
I think this is because corehost.proj adds `#include windows.h`  to the native versions file and now we do not include the file.

---
In reply to: [578795224](https://github.com/dotnet/runtime/pull/48254#discussion_r578795224) [](ancestors = 578795224)

Reply:
Actually we never included that file in singlefilehost, since it is built with `set(SKIP_VERSIONING 1)`, so now I am not sure.
Somehow NTSTATUS is defined when building installer partition and not when building coreclr. 

And that is probably the same reason why `#include <shellapi.h>` is needed to be included in `pal.windows.cpp`. 

---
In reply to: [578819238](https://github.com/dotnet/runtime/pull/48254#discussion_r578819238) [](ancestors = 578819238,578795224)